### PR TITLE
BUG: Having quotes in the description crashes the tool

### LINKED ISSUE #28: BUG: Having quotes in the description crashes the tool
See [this answer](https://stackoverflow.com/a/65878993/1435116).

## Example

On _Copa Coupona_, executing this command would fail, because the summary of the issue contains `Can't` with a single quote.

```bash
pr -i 361
```

## Acceptance Criteria

- The tool no longer fails when pull request summary contains single or double quotes.

## Shaping

- Every time a CLI is invoke with dynamic content (from GH or from the command line options)
  - replace every `"` with `\x22`
  - replace every `'` with `\x27`

### PR DESCRIPTION
**Pull request by jmarsh24**
fixes #28

### CODE REVIEW

**Review by danieldiekmeier — DISMISSED** [review]

File: exe/pr [hunk 1]
```diff
@@ -84,4 +85,5 @@ sh "git push -u origin #{branch}"
 
 puts "Creating pull request"
 ENV["PR_BODY"] = body_lines.join("\n")
-sh "gh pr create --draft --title '#{title}' --body \"$PR_BODY\" --base #{base_branch}"
+
+sh "gh pr create --draft --title #{Shellwords.escape(title)} --body #{Shellwords.escape(body)} --base #{base_branch}"
```
Comment (JensRavens) [inline exe/pr:89]: if this actually works for multiline strings (and during my experiments it did not), then the line above is not needed anymore

**Review by JensRavens — CHANGES_REQUESTED** [review]

File: exe/pr (hunk 1 above)
Comment (Adeynack) [reply exe/pr]: @JensRavens @jmarsh24 : Was that resolved?

**Review by Adeynack — CHANGES_REQUESTED** [review]

File: exe/pr (hunk 1 above)
Comment (JensRavens) [reply exe/pr]: that's exactly the line you requested an RFC, so it's still open

File: exe/pr [hunk 2]
```diff
@@ -77,11 +76,16 @@ sh "git checkout -b #{branch}"
...
 sh "git push -u origin #{branch}"
 
 puts "Creating pull request"
-ENV["PR_BODY"] = body_lines.join("\n")
-sh "gh pr create --draft --title '#{title}' --body \"$PR_BODY\" --base #{base_branch}"
+
+escaped_title = title.gsub("'", "\\\\'")
+escaped_body = body_lines.join("\n").gsub("'", "\\\\'")
+
+ENV["PR_BODY"] = escaped_body
```
[the remark this answers is not in the document]
Comment (Adeynack) [reply exe/pr:89]: We just tested @jmarsh24 and I and it does not need to be in an ENV.

```ruby
sh "gh pr create --draft --title \"#{escaped_title}\" --body '#{escaped_body}' --base #{base_branch}"
```

**Review by Adeynack — CHANGES_REQUESTED** [review]

**Review by JensRavens — APPROVED** [review]

**Pull request merged**